Introducing Gamemodes

fiteclub opened the pull request:
![Screen Shot 2020-12-02 at 7 17 45 AM](https://user-images.githubusercontent.com/3578126/100871603-88bbfb80-346e-11eb-8019-98c763a08ac2.png)


This PR introduces a title screen with multiple game modes.  ~~It can be viewed at /warrior/pvp.~~
- One Player : manual entry vs a random warrior.
- Two Player : manual entry vs manual entry.
- CPU Only : random warrior vs random warrior. (current default)

~~"pvp" is the tentative term used to describe interaction with manual entries, as in One Player and Two Player modes.~~

Other QOL improvements:
- ```droid-say.html.erb``` to display messages in the footer
- Geographic autocomplete 
- Minor UI fixes

todtb commented on app/assets/stylesheets/application.scss:
```diff
@@ -30,8 +30,6 @@ body, html {
 }
 
 .bg.duel{
-	/* The image used */
-	/* Taken frrom https://sweet-cider.tumblr.com/post/161767766015 */
```
These types of comments are never a bad thing for a public site. Most Creative Commons licenses (and other licensing as well) allow free use with attribution to creator/owner. That said, is this image ('mustafar') licensed for use?

fiteclub replied:
This is actually a legacy comment, as the linked background is no longer used, so that can be removed.  The mustafar image is a screenshot from the [Episode III GBA version](https://starwars.fandom.com/wiki/Star_Wars_Episode_III:_Revenge_of_the_Sith_(video_game)), so I'm not sure about licensing.

todtb commented on app/controllers/warriors_controller.rb:
```diff
@@ -61,6 +61,15 @@ def destroy
     end
   end
 
+  def pvp
```
Often times and empty method is written shorthand:

```ruby
def pvp; end
```

fiteclub replied:
That's visually pleasing! Fixed.

todtb commented on app/views/warriors/_tie.html.erb:
```diff
@@ -11,3 +15,6 @@
 </p>
+<br>
+<br>
+<%= render partial: 'droid_say', locals: { droid_message: "Sir, the odds of both players having an elevation of exactly #{number_with_delimiter(@player_one.elevation.round(2), :precision => 2, :delimeter=> ',')} meters is approximately three-thousand-seven-hundred-twenty to one!" } %>
```
lol, love it

todtb commented on app/views/warriors/_versus.html.erb:
```diff
@@ -7,6 +7,6 @@
       </span>
     </div>
-    <%= render partial: 'player', locals: { player: @player_two, winner: @results[:player_two_win] } %>
+    <%= render partial: 'player', locals: { player: @player_two, winner: @player_two==@results[:winner] } %>
```
Purely a style recommendation:

```ruby
@player_two == @results[:winner]
```

todtb commented on app/views/warriors/pvp.html.erb:
```diff
@@ -0,0 +1,81 @@
+          <div class="row">
+            <div class="col-sm-12 text-center">
+            <br><br>
+
+
+
```
More style: lots of empty lines which can be removed.

todtb commented on config/routes.rb:
```diff
@@ -1,5 +1,6 @@
 Rails.application.routes.draw do
   root 'warriors#duel'
+  get 'warriors/pvp'
   resources :warriors
```
You can also write this:

```ruby
resources :warriors do
  get 'pvp', on: :collection
end
```

https://guides.rubyonrails.org/routing.html

fiteclub replied:
Way cool, agreed!

todtb reviewed (commented):
Looking good @fiteclub!

fiteclub commented:
I was using rails-erb-loader to pass environmental variables (API key) to js.  However, this caused build failures on heroku. Removed rails-erb-loader entirely, and refactored using rails credentials and webpacker.rb initializer.

This PR is now live on heroku: [https://hidden-inlet-56466.herokuapp.com/](https://hidden-inlet-56466.herokuapp.com/)
